django/db/models/fields/composite.py

```diff
@@ -130,6 +137,21 @@ def _check_field_name(self):
             )
         ]
 
+    def value_to_string(self, obj):
+        values = []
+        vals = self.value_from_object(obj)
+        for field, value in zip(self.fields, vals):
+            obj = AttributeSetter(field.attname, value)
+            values.append(field.value_to_string(obj))
+        return json.dumps(values, ensure_ascii=False)
+
+    def to_python(self, value):
+        if isinstance(value, str):
+            # Assume we're deserializing.
+            vals = json.loads(value)
+            value = [field.to_python(val) for field, val in zip(self.fields, vals)]
```

Comment:
Should we raise if the PK contains extra garbage?
```suggestion
            value = [
                field.to_python(val)
                for field, val in zip(self.fields, vals, strict=True)
            ]
```

Reply:
Thank you! Updated and added a test :+1: